EIPS/eip-7144.md

```diff
@@ -0,0 +1,345 @@
+---
+eip: 7144
+title: ERC-20 with transaction validation step.
+description: A new validation step for transfer and approve calls, achieving a security step in case of stolen wallet.
+author: Eduard López i Fina (@eduardfina)
+discussions-to: https://ethereum-magicians.org/t/erc721-with-a-validation-step/14071
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-07
+requires: 20
+---
+
+## Abstract
+
+This standard is an extension of [ERC-20](./eip-20.md). It defines new validation functionality to avoid wallet draining: every `transfer` or `approve` will be locked waiting for validation.
+
+## Motivation
+
+The power of the blockchain is at the same time its weakness: giving the user full responsibility for their data.
+
+Many cases of Token theft currently exist, and current Token anti-theft schemes, such as transferring Tokens to cold wallets, make Tokens inconvenient to use.
```

Comment:
it's challenging for me to understand how adding an interface to record "valid/invalid transaction" can help reduce token theft. Can you elaborate with concrete and specific examples?

Reply:
Good question @xinbenlv 

The transfer and approve transactions will generate a new pending validation. Then with `_validateTransfer` and `_validateApproval` the validations can be accepted.

The standard doesn't defines who can call the validation functions but I did it for the chainlink hackathon, if you have 5 minutes maybe the presentation video can be useful to understand the standard. :)

[Securechain video](https://www.youtube.com/watch?v=v-rs1eO-mZs)

Reply:
What's people's incentive to pay for gas fee to validate transaction that will be recorded onchain?

Reply:
If someone have your private key will not be able to steal your tokens/NFTs because will need to validate the transfer.

Is a security method, sometimes you want to have your assets in your hot wallet (f.e. Metamask) so you can use then in some dApps (DAOs, Marketplaces, etc.)

You'll be able to use it without the risk of wallet draining.

For expensive NFTs like bored apes or big quantity of tokens the cost of validation is minimum in comparison with the risk you are avoiding.

Reply:
To do a transfer first you'll need to call the transfer transaction and then validate it using the dApp (in my example).